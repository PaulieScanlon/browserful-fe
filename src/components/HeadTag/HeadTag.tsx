import * as React from 'react';
import Head from 'next/head';

export const HeadTag: React.SFC = () => {
  return (
    <Head>
      <title>Browserful</title>
    </Head>
  );
};
