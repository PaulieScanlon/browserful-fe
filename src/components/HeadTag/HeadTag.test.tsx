import * as React from 'react';
import { shallow } from 'enzyme';

import { HeadTag } from '../HeadTag';

const shallowDefault = () => {
  return shallow(<HeadTag />);
};

describe('<HeadTag />', () => {
  it('renders correctly', () => {
    const wrapper = shallowDefault();
    expect(wrapper).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });
  it('has a <title>', () => {
    const wrapper = shallowDefault();
    const title = wrapper.find('title');
    expect(title.length).toEqual(1);
  });
});
