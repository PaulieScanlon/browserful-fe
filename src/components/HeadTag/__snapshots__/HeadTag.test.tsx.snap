// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeadTag /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HeadTag />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <title>
          Browserful
        </title>,
        <meta
          content="Last 5 versions, > 0.1%, not dead. This is confusing. Browserful isn’t!"
          property="og:title"
        />,
        <meta
          content="http://www.browserful.com/static/images/opengraph.jpg"
          property="og:image"
        />,
        <meta
          content="Using our easy to use sliders you can configure a matrix to show exactly which browsers you support and which browsers you don’t."
          property="og:description"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Browserful",
        },
        "ref": null,
        "rendered": "Browserful",
        "type": "title",
      },
      Object {
        "instance": null,
        "key": "og:title",
        "nodeType": "host",
        "props": Object {
          "content": "Last 5 versions, > 0.1%, not dead. This is confusing. Browserful isn’t!",
          "property": "og:title",
        },
        "ref": null,
        "rendered": null,
        "type": "meta",
      },
      Object {
        "instance": null,
        "key": "og:image",
        "nodeType": "host",
        "props": Object {
          "content": "http://www.browserful.com/static/images/opengraph.jpg",
          "property": "og:image",
        },
        "ref": null,
        "rendered": null,
        "type": "meta",
      },
      Object {
        "instance": null,
        "key": "og:description",
        "nodeType": "host",
        "props": Object {
          "content": "Using our easy to use sliders you can configure a matrix to show exactly which browsers you support and which browsers you don’t.",
          "property": "og:description",
        },
        "ref": null,
        "rendered": null,
        "type": "meta",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <title>
            Browserful
          </title>,
          <meta
            content="Last 5 versions, > 0.1%, not dead. This is confusing. Browserful isn’t!"
            property="og:title"
          />,
          <meta
            content="http://www.browserful.com/static/images/opengraph.jpg"
            property="og:image"
          />,
          <meta
            content="Using our easy to use sliders you can configure a matrix to show exactly which browsers you support and which browsers you don’t."
            property="og:description"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Browserful",
          },
          "ref": null,
          "rendered": "Browserful",
          "type": "title",
        },
        Object {
          "instance": null,
          "key": "og:title",
          "nodeType": "host",
          "props": Object {
            "content": "Last 5 versions, > 0.1%, not dead. This is confusing. Browserful isn’t!",
            "property": "og:title",
          },
          "ref": null,
          "rendered": null,
          "type": "meta",
        },
        Object {
          "instance": null,
          "key": "og:image",
          "nodeType": "host",
          "props": Object {
            "content": "http://www.browserful.com/static/images/opengraph.jpg",
            "property": "og:image",
          },
          "ref": null,
          "rendered": null,
          "type": "meta",
        },
        Object {
          "instance": null,
          "key": "og:description",
          "nodeType": "host",
          "props": Object {
            "content": "Using our easy to use sliders you can configure a matrix to show exactly which browsers you support and which browsers you don’t.",
            "property": "og:description",
          },
          "ref": null,
          "rendered": null,
          "type": "meta",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
